packages/web/src/pages/upload-page/fields/availability/usdc-purchase-gated/UsdcPurchaseGatedRadioField.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+  FeatureFlags,
+  PremiumConditions,
+  TrackAvailabilityType,
+  isPremiumContentUSDCPurchaseGated
+} from '@audius/common'
+import { IconCart, IconStars } from '@audius/stems'
+
+import { ExternalLink } from 'components/link'
+import { ModalRadioItem } from 'components/modal-radio/ModalRadioItem'
+import { useFlag } from 'hooks/useRemoteConfig'
+
+import { UsdcPurchaseFields } from './UsdcPurchaseFields'
+import styles from './UsdcPurchaseGatedRadioField.module.css'
+
+const WAITLIST_TYPEFORM = 'https://example.com'
+
+const messages = {
+  usdcPurchase: 'Premium (Pay-to-Unlock)',
+  usdcPurchaseSubtitle:
+    'Unlockable by purchase, these tracks are visible to everyone but only playable by users who have paid for access.',
+  waitlist:
+    'Start selling your music on Audius today! Limited access beta now available.',
+  join: 'Join the Waitlist',
+  comingSoon: 'Coming Soon'
+}
+
+type UsdcPurchaseGatedRadioFieldProps = {
+  isUpload?: boolean
+  initialPremiumConditions?: PremiumConditions
+  isInitiallyUnlisted?: boolean
+}
+
+export const UsdcPurchaseGatedRadioField = (
+  props: UsdcPurchaseGatedRadioFieldProps
+) => {
+  const { isUpload, initialPremiumConditions, isInitiallyUnlisted } = props
+
+  const { isEnabled: isUsdcUploadEnabled } = useFlag(
+    FeatureFlags.USDC_PURCHASES_UPLOAD
+  )
+
+  const noUsdcPurchase =
+    !isUpload &&
+    !isPremiumContentUSDCPurchaseGated(initialPremiumConditions) &&
+    !isInitiallyUnlisted
+
+  const disabled = noUsdcPurchase || !isUsdcUploadEnabled
```

Comment:
nit: why not roll this up into the above?

Reply:
i think this is more self-documenting, b/c i borrowed this logic from elsewhere - especially since it's used in edit too.